akka-docs/src/main/paradox/typed/persistence-lifecycle.md

```diff
@@ -0,0 +1,91 @@
+  
+### Snapshot deletion
+
+To free up space, an event sourced actor can automatically delete older snapshots 
+if there are saved snapshots which match the `SnapshotSelectionCriteria`. 
+Deletion of snapshots is currently based on a configured or default `RetentionCriteria` 
+from `withRetention`combined with either the `snapshotWhen` or `snapshotEvery` methods. 
+If snapshots are enabled, the deletion sliding window leverages the `RetentionCriteria` to construct
+a `SnapshotSelectionCriteria` based on the latest sequence number. 
+ 
+@@@ note
+
+[This API will be simplified soon](https://github.com/akka/akka/issues/26544).
+
+@@@ note
+  
+Scala
+:  @@snip [BasicPersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala) { #snapshotDeletes }
+
+Java
+:  @@snip [BasicPersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BasicPersistentBehaviorTest.java) { #snapshotDeletes }
+
+On async deletion, either a `SnapshotCompleted` or `SnapshotFailed` is returned. Successful completion is logged by the system at log level `debug`, failures at log level `error`.
+You can leverage `EventSourcedSignal` to react to outcomes with `EventSourcedBehavior.receiveSignal`.
+
+Scala
+:  @@snip [BasicPersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala) { #fullDeletesSampleWithSignals }
+
+Java
+:  @@snip [BasicPersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BasicPersistentBehaviorTest.java) { #snapshotDeletesSifullDeletesSampleWithSignalsgnal }
+
+## Event deletion
+If snapshot-based recovery is enabled, you can elect to have the system automatically delete older events by enabling `RetentionCriteria.deleteEventsOnSnapshot`, which is disabled by default. 
+Additionally you can leverage `EventSourcedSignal` to react to outcomes with `EventSourcedBehavior.receiveSignal`.
+   
+Scala
+:  @@snip [BasicPersistentBehaviorCompileOnly.scala](/akka-persistence-typed/src/test/scala/docs/akka/persistence/typed/BasicPersistentBehaviorCompileOnly.scala) { #snapshotAndEventDeletes }
+
+Java
+:  @@snip [BasicPersistentBehaviorTest.java](/akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/BasicPersistentBehaviorTest.java) { #snapshotAndEventDeletes }
+ 
+On `SaveSnapshotSuccess`, old events would be deleted based on `RetentionCriteria` prior to old snapshots being deleted. On async deletion, either `DeleteEventsCompleted` or `DeleteEventsFailed` is returned. Successful completion is logged by the 
+system at log level `debug`, failures at log level `error`. You can leverage `EventSourcedSignal` to react to outcomes with `EventSourcedBehavior.receiveSignal`.
+
+Message deletion does not affect the highest sequence number of the journal, even if all messages were deleted from it after a delete occurs.
```

Comment:
Did persistence grow to long or why did you factor this out? I think the deletion logically fits nicely in the snapshotting docs rather and the warning about side effects on recovery in the main persistence doc?

Reply:
I think this is part of what I originally took from the untyped event delete doc and left here for discussion during review more than anything.

Reply:
Oh this came from the ticket where I suggested the lifecycle would be a nice breakout because persistence.md alone is all concepts and this gets more into the lifecycle which for a user is really helpful. I'm happy to keep as one doc but the organization of the persistence.md seemed lacking.